src/game_state.rs

```diff
@@ -3,14 +3,38 @@ use rand;
 use geometry::{Position, Size};
 use models::World;
 
+/// This is a message that will be drawn to the screen. When it's shown on the screen the game
+/// will be waiting for user input
+pub struct Message {
+    pub title: &'static str,
+    pub subtitle: &'static str,
+    pub should_show: bool
```

Comment:
Having `should_show` here feels like a hack... It is not really part of a message. See my comment to line 35 for a suggestion.